hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2013-2017 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.  See the License for the specific language
+# governing permissions and limitations under the License.
+"""A switch driver for OpenVswitch.
+"""
+import re
+import logging
+import schema
+import os
+import ast
+import subprocess
+import shlex
+
+from hil.model import db, Switch, BigIntegerType, SwitchSession
+from hil.errors import SwitchError
+
+logger = logging.getLogger(__name__)
+
+# Class layout
+# 1. Public methods:
+# 2. Private methods
+# 3. Unimplemented superclass methods.
+
+
+class Ovs(Switch, SwitchSession):
+    """ Driver for openvswitch. """
+    api_name = 'http://schema.massopencloud.org/haas/v0/switches/ovs'
+    dir_name = os.path.dirname(__file__)
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+    }
+
+    id = db.Column(
+            BigIntegerType, db.ForeignKey('switch.id'), primary_key=True
+            )
+    hostname = db.Column(db.String, nullable=False)
+    username = db.Column(db.String, nullable=False)
+    password = db.Column(db.String, nullable=False)
+
+    @staticmethod
+    def validate(kwargs):
+        """Checks input to match switch parameters."""
+        schema.Schema({
+            'username': basestring,
+            'hostname': basestring,
+            'password': basestring,
+        }).validate(kwargs)
+
+    @staticmethod
+    def validate_port_name(port):
+        """ This driver accepts any string as port name."""
+
+# 1. Public Methods:
+
+    def get_capabilities(self):
+        """Provides the features of this switch that HIL supports. """
+        return['Virtual_switch_for_development_Purpose_only']
```

Comment:
Make it `return []` or `return ['nativeless-trunk-mode']` if it supports that.

Reply:
I understand that `get_capabilities` should return info about what the switch can do. 
I thought this comment is appropriate since it is just like any generic L2 switch but virtual and to be used only for development purpose. I felt that it is more important to emphasize that it should not be used for any production setup. Its **capability** is that it is for development purpose only.

Reply:
yeah, but it's not for the end users to see. Only an API exposing a certain call (enable jumbo frame, qos etc) would call this method and look for that string.

 Putting that string as it's capability does nothing to emphasize that it's for production. Best way to emphasize that is to put it in the docs, not this method. It's not meant to be a comment box.

Reply:
I understand now. In that case I will return an empty list. next commit fixes it.